### ISSUE TITLE
[Question] How to plot prediction in tensorboard for torch/lightning models?

### ISSUE THREAD

**Issue by AlexVeuthey** [unit8co/darts#2402]
I'm trying to log graphs/plots of the predictions on the validation set during training a PyTorch-Lightning based model, but cannot find an easy and efficient way. I've looked in the doc and issues without success. The goal is to monitor during training if the model is converging well enough on actual data, rather than rely on metrics only.

My approach was to write a custom Lightning callback, and use its `.on_validation_batch_end()` method to recover the outputs of the first validation batch, store them in a callback member, and then plot them to tensorboard using `.on_validation_epoch_end()`.

However, the `outputs` argument of `.on_validation_batch_end()` is not populated (empty tensor), thus requiring a manual step to recover the predictions from the model and the `batch` argument. Also, I'm unsure why, but the `batch` argument does not seem to contain the future (target) values to compare the predictions against, which makes this approach not very useful...

Is there another way to achieve this?
